src/filters/truncate.js

```diff
@@ -2,30 +2,33 @@
 
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
-    return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+    return function truncate(str, charLimit, useWordBoundary, newlineLimit, removeExtraChars) {
+      if(typeof str !== 'string' || typeof charLimit !== 'number') {
         return str;
       }
 
       var truncated = str;
+      var limiter = charLimit;
+      var removedCharsLength = charLimit - (removeExtraChars || 0);
 
-      if (charLimit) {
-        truncated = truncated.substring(0, charLimit);
-      }
-
-      if (newlineLimit) {
+      if (newlineLimit && truncated.length > limiter) {
```

Comment:
The intent was to make newlineLimit a hard limit, even if the string is < char limit. `foo\nbar` should get truncated still if I say newline limit 1, char limit 80, even though it's less than 80 characters.